sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -609,6 +611,12 @@ class SparkSession private(
    */
   @transient lazy val catalog: Catalog = new CatalogImpl(self)
 
+  @transient private lazy val catalogs = new mutable.HashMap[String, CatalogProvider]()
+
+  private[sql] def catalog(name: String): CatalogProvider = synchronized {
```

Comment:
Note that this is `private[sql]`. This allows us to use the named `TableCatalog` instances without solving how multiple catalogs should be exposed to users through a public API just yet.